File: orbees-api/Models/Category.cs
using Api.Models.Common;

namespace Api.Models
{
    public class Category : AuditableEntity
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public required string Name { get; set; }
        public string? Icon { get; set; }
        public string? Color { get; set; }
        public bool IsActive { get; set; } = true;

        // FK USER -- null se for categoria padrão
        public Guid? UserId { get; set; }
        public User? User { get; set; }

        // FK Grupo -- null se for individual
        public Guid? GroupId { get; set; }
        public Group? Group { get; set; }
    }
}

File: orbees-frontend/src/interfaces/category.ts
export interface CategoryReadDto {
  id: string;
  name: string;
  icon?: string;
  color?: string;
  isSystemCategory: boolean;
  groupId?: string;
  groupName?: string;
  isSystem: boolean;
}

export interface CategoryCreateDto {
  name: string;
  icon?: string;
  color?: string;
  groupId?: string;
}

export interface CategoryUpdateDto {
  name?: string;
  icon?: string;
  color?: string;
}
